pyaedt/icepak.py

```diff
@@ -4625,3 +4625,266 @@ def _create_node_dict(self, default_dict):
                         node_args[k] = val
 
                 return node_args
+
+    @pyaedt_function_handler
+    def assign_solid_block(
+        self, object_name, power_assignment, boundary_name=None, htc=None, ext_temperature="AmbientTemp"
+    ):
+        """
+        Assign block boundary for solid objects.
+
+        Parameters
+        ----------
+        object_name : str
+            Name of the object.
+        power_assignment : str or dict
+            String with the value and units of the power assignment or with
+            ``"Joule Heating"``. For a temperature-dependent or transient
+            assignment, a dictionary can be used. The dictionary should contain three keys:
+            ``"Type"``, ``"Function"``, and ``"Values"``.
+
+            - For the ``"Type"`` key, accepted values are ``"Temp Dep"`` and ``"Transient"``.
+            - For the ``"Function"`` key, acceptable values depend on the ``"Type"`` key
+              selection. When the ``"Type"`` key is set to ``"Temp Dep"``, the only
+              accepted value is ````"Piecewise Linear"`` . When the ``"Type"`` key is
+              set to ``"Transient"``, acceptable values are `"Exponential"``, `"Linear"``,
+              ```"Piecewise Linear"``, ``"Power Law"``, ``"Sinusoidal"``, and ``"Square
+               Wave"``.
+            - For the ``"Values"`` key, a list of strings contain the parameters required by
+              the ``"Function"`` key selection. For example, whn``"Linear"`` is set as the
+              ``"Function"`` key, two parameters are required: the value of the variable
+              at t=0 and the slope of the line. For the parameters required by each
+              ``"Function"`` key selection, see the Icepack documentation (). The parameters
+              must contain the units where needed.
+
+        boundary_name : str, optional
+            Name of the source boundary. The default is ``None``, in which case the
+            boundary name is automatically generated.
+        htc : float, str, or dict, optional
+            String with the value and units of the heat transfer coefficient for the
+            external conditions. If a float is provided, ``"w_per_m2kel"`` unit will be used.
```

Comment:
```suggestion
            external conditions. If a float is provided, ``"w_per_m2kel"`` unit is used.
```
Avoid future tense. Always use present tense when possible.